### PR TITLE
feat: Build footer

### PR DESCRIPTION
**Pull request by jsieving**
### 🔗 Linked issue

https://energysage.atlassian.net/browse/CED-376

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- Added `EsFooter` molecule to the design system
- I tried to avoid using custom styles, so there are a couple places I compromised on the designs, e.g. having a 3rem margin off by 4px
- Smaller SVGs are hard-coded in as specified in the ticket, but at some point we may want to extract them as in https://github.com/EnergySage/es-ds/pull/769.
- One SVG was unnecessarily detailed and made up about 3/4 of the compiled HTML component, so I replaced it with the PNG used in the existing footer.
- Some URLs still need to be added, as the pages need to be verified or don't exist yet.

### 🥼 Testing
- Manual testing on demo server and on local dev environment of es-site
- There is a basic Jest test, but I'm not sure if it needs further testing since it's a static component

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.
- [ ] I have documented testing approach

### CODE REVIEW

**Comment by jimlind** [comment]
Incredibly quick turn around and following your intuition to fill in the gaps between mocked sizes makes the responsive footer work great.

In the mock the social icons are always left aligned but in your version they go full aligned.
Both are valid choices but we should follow the mocks here.

Screen Shot 2023-04-18 at 9 53 38 AM
Screen Shot 2023-04-18 at 9 53 44 AM

File: es-vue-base/src/lib-components/EsFooter.vue
Comment (mpleroux) [inline es-vue-base/src/lib-components/EsFooter.vue:31]: I'm concerned about the IDs and URLs in the SVGs.

It's a pain to hand-edit SVG code, but since these footer SVGs will be included in every page we should [prefix them](https://github.com/EnergySage/es-site/blob/develop/docs/frontend/svg.md#process) (with something like `footer-`?) to avoid conflicting with other SVGs.

I think it's inevitable that some other exported SVG will use the same `#a`. What do you think?

**Review by mpleroux — COMMENTED** [review]
I assume a component more complex than a typical molecule should be moved to `organisms` eventually.

Should the "Make an impact" section at the top of the Figma designs be part of this ticket?

File: es-vue-base/src/lib-components/EsFooter.vue
Comment (RyanMulready) [inline es-vue-base/src/lib-components/EsFooter.vue:31]: Rule of thumb, svg's should only be inlined if they are iconography. If they are in fact real images they are better served as an normal asset using `b-img src="mySvg.svg"`

File: es-vue-base/src/lib-components/EsFooter.vue
Comment (RyanMulready) [inline es-vue-base/src/lib-components/EsFooter.vue:50]: Could these other links/copy also be part of the `data()` response?

File: es-vue-base/src/lib-components/EsFooter.vue
Comment (jimlind) [reply es-vue-base/src/lib-components/EsFooter.vue:31]: This specific SVG is the EnergySage typeface+leaf which doesn't exist in the es-ds yet but will after that gets merged. I was thinking it would be a non-blocking fast follow to replace with the actual logo but could prioritize it if that makes sense: https://energysage.atlassian.net/browse/CED-413

Comment (jimlind) [reply es-vue-base/src/lib-components/EsFooter.vue:31]: This specific SVG is the EnergySage typeface+leaf which doesn't exist in the es-ds yet but will after that gets merged. I was thinking it would be a non-blocking fast follow to replace with the actual logo but could prioritize it if that makes sense: https://energysage.atlassian.net/browse/CED-413

File: es-vue-base/src/lib-components/EsFooter.vue
Comment (RyanMulready) [inline es-vue-base/src/lib-components/EsFooter.vue:76]:
```suggestion
            © Copyright 2009-{{ new Date().getFullYear() }} EnergySage, Inc. All rights reserved.
```

File: es-vue-base/tests/EsFooter.spec.js
Comment (RyanMulready) [inline es-vue-base/tests/EsFooter.spec.js:7]:
```suggestion
describe('EsFooter', () => {
    // Basic test; should exist for most components
    test('<EsFooter />', async () => {
```

File: es-vue-base/tests/__snapshots__/EsFooter.spec.js.snap
Comment (RyanMulready) [inline es-vue-base/tests/__snapshots__/EsFooter.spec.js.snap:3]:
```suggestion
exports[`EsFormMsg <EsFooter /> 1`] = `
```

File: es-vue-base/src/lib-components/EsFooter.vue
Comment (jimlind) [reply es-vue-base/src/lib-components/EsFooter.vue:50]: Up to Jane if this is a quick fix. These links should be changing even less than the ones in the data method so building that logic feels very optional.

Comment (jsieving) [reply es-vue-base/src/lib-components/EsFooter.vue:50]: If we factored out the social icons I think this would make sense, but I'm inclined to leave it as is. _If_ these change, it feels like extracting just the names and URLs will make it more likely that we end up with mismatched links & icons.

**Comment by jsieving** [comment]
> Both are valid choices but we should follow the mocks here.

Bye-bye, lack of custom styles 🥲 

> Should the "Make an impact" section at the top of the Figma designs be part of this ticket?

Apparently it is... I hallucinated a comment that I was supposed to ignore it. Will do.

File: es-vue-base/src/lib-components/EsFooter.vue
Comment (RyanMulready) [reply es-vue-base/src/lib-components/EsFooter.vue]: I will note the file does exist in the [navigation](https://github.com/EnergySage/es-ds/pull/769/files#diff-4bd3f38780247990e68745676ee468f510b020477d3fedc08a1ce71f573e2c92) branch. The DOE logo should get similar treatment.

Comment (jsieving) [reply es-vue-base/src/lib-components/EsFooter.vue]: We're using the PNG from the current site footer for the DOE logo because the SVG for that logo is ridiculously detailed/oversized and still a little broken. I can copy over the ES logo from the other branch, but it seems simpler to do that after one of these PRs is merged.

**Comment by jsieving** [comment]
I think this is ready for final review.

**Review by RyanMulready — APPROVED** [review]
This looks great. 🔥  I appreciate the quick turn around on suggestions/feedback. ❤️

**Review by tomleo — CHANGES_REQUESTED** [review]
The feature looks great, however I might suggest upgrading `es-footer` to an *organism* vs. a *molecule*.

I'm also not sure if the footer should be packaged in `es-vue-base`, I think having it in the `es-design-system` should be enough.

Here's what I was thinking https://github.com/EnergySage/es-ds/pull/779

**Review by tomleo — APPROVED** [review]
👍

File: es-vue-base/src/lib-components/EsFooter.vue
Comment (ericdouglaspratt) [inline es-vue-base/src/lib-components/EsFooter.vue:110]: Optional: In cases where we care about preserving the order of a set of items in JavaScript, I'd generally still recommend using an array, even though in most modern browser implementations, the order of chronological definition will actually be preserved and this will work. So not a necessary change but I did change the nav data objects to be arrays just for extra precaution.
https://www.stefanjudis.com/today-i-learned/property-order-is-predictable-in-javascript-objects-since-es2015/

File: es-design-system/pages/atoms/icons.vue
Comment (ericdouglaspratt) [inline es-design-system/pages/atoms/icons.vue:383]: Nit: I'd love if we could eventually get 32px by 32px assets from design so these can match the other icons in being an exact square, but that can be done later.

File: es-design-system/pages/molecules/es-footer.vue
Comment (ericdouglaspratt) [inline es-design-system/pages/molecules/es-footer.vue:7]: Optional: Would be great if the docs page used a layout where the footer could go full-width so we can give it a proper layout test, but we can also do that later.

**Review by ericdouglaspratt — APPROVED** [review]
Looks good! A few small things we may want to do in the future, but nothing that needs to block merging this in now.

**Pull request merged**
